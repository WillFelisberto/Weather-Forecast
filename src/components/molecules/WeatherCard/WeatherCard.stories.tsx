import type { Meta, StoryObj } from '@storybook/react';

import { WeatherCard } from './WeatherCard';

const meta: Meta<typeof WeatherCard> = {
  title: 'Molecules/WeatherCard',
  component: WeatherCard,
  tags: ['autodocs'],
  args: {
    location: 'New York',
    temperature: 25,
    windSpeed: 15,
    datatime: '2025-02-04T19:15',
    weatherCode: 0
  },
  argTypes: {
    location: { control: 'text' },
    temperature: { control: 'number' },
    windSpeed: { control: 'number' },
    datatime: { control: 'text' },
    weatherCode: {
      control: 'select',
      options: [
        0, 1, 2, 3, 45, 48, 51, 53, 55, 61, 63, 65, 66, 67, 71, 73, 75, 80, 81, 82, 95, 96, 99
      ]
    }
  }
};

export default meta;

type Story = StoryObj<typeof WeatherCard>;

export const Default: Story = {};

export const PartlyCloudy: Story = {
  args: {
    location: 'London',
    temperature: 18,
    windSpeed: 10,
    datatime: '2025-02-04T19:15',
    weatherCode: 2
  }
};

export const Thunderstorm: Story = {
  args: {
    location: 'SÃ£o Paulo',
    temperature: 22,
    windSpeed: 20,
    datatime: '2025-02-04T19:15',
    weatherCode: 80
  }
};
