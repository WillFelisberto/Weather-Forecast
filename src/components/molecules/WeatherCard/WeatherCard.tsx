import { clsx } from 'clsx';
import React from 'react';

import { getWeatherIconSVG } from '@/lib/utils/icons';

export interface WeatherCardProps {
  location: string;
  temperature: number;
  windSpeed: number;
  datatime: string;
  weatherCode: number;
  className?: string;
}

export const WeatherCard: React.FC<WeatherCardProps> = ({
  location,
  temperature,
  windSpeed,
  datatime,
  weatherCode,
  className
}) => {
  const { weatherCondition, weatherIcon } = getWeatherIconSVG(
    weatherCode,
    weatherCode !== 0 ? 300 : 400
  );

  const dayOfWeek = new Date(datatime).toLocaleDateString('en-US', { weekday: 'long' });
  const timeFormatted = new Date(datatime).toLocaleTimeString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: false
  });

  return (
    <div
      className={clsx(
        'bg-gradient-to-tr from-[--card-bg] to-[--card-bg-gradient]   min-w-[355px] px-[52px] py-[40px] text-left justify-end h-auto text-white p-6 rounded-2xl shadow-lg flex flex-col',
        className
      )}
    >
      <div>{weatherIcon}</div>
      <div className=" leading-[65px] flex flex-row center gap-[8px]">
        <h2 className="text-3xl truncate leading-[65px] font-secondary font-bold capitalize">
          {location}
        </h2>
        <p className="text-3xl leading-[65px] font-secondary font-light  ">{temperature}Â°C</p>
      </div>

      <p className="text-2xl mt-[0px] leading-[65px] font-secondary font-light">
        Wind Speed: {windSpeed} km/h
      </p>
      <p className="text-2xl mt-[0px] leading-[65px] font-secondary font-light">{`${dayOfWeek}: ${timeFormatted}`}</p>
      <p className="capitalize mt-[0px] leading-[65px] text-2xl capitalize font-secondary font-light ">
        {weatherCondition}
      </p>
    </div>
  );
};
