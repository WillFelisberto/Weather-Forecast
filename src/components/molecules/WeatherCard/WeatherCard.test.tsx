import { render, screen } from '@/tests/test-utils';
import React from 'react';

import '@testing-library/jest-dom';
import { WeatherCard } from './WeatherCard';

// Mock da função getWeatherIconSVG
jest.mock('@/lib/utils/icons', () => ({
  getWeatherIconSVG: jest.fn(() => ({
    weatherCondition: 'Cloudy',
    weatherIcon: <svg data-testid="weather-icon" />
  }))
}));

describe('WeatherCard Component', () => {
  const defaultProps = {
    location: 'Tehran',
    temperature: 35,
    windSpeed: 11,
    datatime: '2025-02-04T19:15',
    weatherCode: 800
  };

  it('should render the location and temperature correctly', () => {
    const { container } = render(<WeatherCard {...defaultProps} />);
    expect(screen.getByText('Tehran')).toBeInTheDocument();
    expect(screen.getByText('35°C')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the wind speed correctly', () => {
    render(<WeatherCard {...defaultProps} />);
    expect(screen.getByText('Wind Speed: 11 km/h')).toBeInTheDocument();
  });

  it('should render the correct day of the week and formatted time', () => {
    render(<WeatherCard {...defaultProps} />);
    expect(screen.getByText('Tuesday: 19:15')).toBeInTheDocument(); // Baseado na data fornecida
  });

  it('should render the weather condition correctly', () => {
    render(<WeatherCard {...defaultProps} />);
    expect(screen.getByText('Cloudy')).toBeInTheDocument();
  });

  it('should display the weather icon', () => {
    render(<WeatherCard {...defaultProps} />);
    expect(screen.getByTestId('weather-icon')).toBeInTheDocument();
  });
});
