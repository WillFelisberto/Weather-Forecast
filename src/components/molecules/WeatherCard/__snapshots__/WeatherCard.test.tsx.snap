// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard Component should render the location and temperature correctly 1`] = `
<div>
  <div
    class="bg-gradient-to-tr from-[--card-bg] to-[--card-bg-gradient]   min-w-[355px] px-[52px] py-[40px] text-left justify-end h-auto text-white p-6 rounded-2xl shadow-lg flex flex-col"
  >
    <div>
      <svg
        data-testid="weather-icon"
      />
    </div>
    <div
      class=" leading-[65px] flex flex-row center gap-[8px]"
    >
      <h2
        class="text-3xl truncate leading-[65px] font-secondary font-bold capitalize"
      >
        Tehran
      </h2>
      <p
        class="text-3xl leading-[65px] font-secondary font-light  "
      >
        35
        Â°C
      </p>
    </div>
    <p
      class="text-2xl mt-[0px] leading-[65px] font-secondary font-light"
    >
      Wind Speed: 
      11
       km/h
    </p>
    <p
      class="text-2xl mt-[0px] leading-[65px] font-secondary font-light"
    >
      Tuesday: 19:15
    </p>
    <p
      class="capitalize mt-[0px] leading-[65px] text-2xl capitalize font-secondary font-light "
    >
      Cloudy
    </p>
  </div>
</div>
`;
